go/vt/sqlparser/sql.y

```diff
@@ -8638,6 +8644,7 @@ non_reserved_keyword:
 | VARIABLES
 | VARIANCE %prec FUNCTION_CALL_NON_KEYWORD
 | VCPU
+| VECTOR
```

Comment:
This follows how the new `VECTOR` type works in MySQL 9.0. It's not a reserved keyword, which means that for example using a column named `vector` that is not escaped in a `SELECT vector FROM t1` still works. 

This also means that the backwards compatibility here is guaranteed, since no working query today would break with this change.